### PR TITLE
sccache: init at 0.2.8

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

This is the beginning of a new compiler cache support for NixOS, starting with the bare package. As discussed here: https://discourse.nixos.org/t/use-sccache-for-rust-compilation-cache/3228 . Currently tests are disabled due to https://github.com/mozilla/sccache/issues/460 .

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) - not applicable.
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"` - non depend.
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/misc/sccache/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 = "08v8s24q6246mdjzl5lirqg0csxcmd17szmw4lw373hvq4xvf0yk";
+  };
+  cargoSha256 = "1lafzin92h1hb1hqmbrsxja44nj8mpbsxhwcjr6rf5yrclgwmcxj";
+  cargoBuildFlags = [ "--features=all" ];
+  nativeBuildInputs = [
+    pkgconfig cargo rustc
+  ];
+  buildInputs = [
+    openssl
+  ];
```
Comment (risicle) [inline pkgs/development/tools/misc/sccache/default.nix]: To build on darwin (tested macos 10.13) this needs `++ stdenv.lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security` added to `buildInputs`

Comment (doronbehar) [reply pkgs/development/tools/misc/sccache/default.nix]: Thanks for testing this! I'm adding this after the closing `]`. Please let me know if that's not what you meant.

**Comment by Mic92** [comment]
@GrahamcOfBorg build sscache

**Comment by Mic92** [comment]
@GrahamcOfBorg build sccache

File: pkgs/development/tools/misc/sccache/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/development/tools/misc/sccache/default.nix]: Perfect.

**Pull request merged**